src/main/java/com/amazon/opendistroforelasticsearch/security/compliance/ComplianceConfig.java

```diff
@@ -105,17 +126,19 @@ private ComplianceConfig(
         this.ignoredComplianceUsersForReadMatcher = WildcardMatcher.from(ignoredComplianceUsersForRead);
         this.ignoredComplianceUsersForWriteMatcher = WildcardMatcher.from(ignoredComplianceUsersForWrite);
         this.opendistrosecurityIndex = opendistrosecurityIndex;
-
-        //opendistro_security.compliance.pii_fields:
-        //  - indexpattern,fieldpattern,fieldpattern,....
-        this.readEnabledFields = watchedReadFields.stream()
-                .map(watchedReadField -> watchedReadField.split(","))
-                .filter(split -> split.length != 0 && !Strings.isNullOrEmpty(split[0]))
-                .collect(ImmutableMap.toImmutableMap(
-                        split -> WildcardMatcher.from(split[0]),
-                        split -> split.length == 1 ?
-                                Collections.singleton("*") : Arrays.stream(split).skip(1).collect(ImmutableSet.toImmutableSet())
-                ));
+        this.watchedReadFields = watchedReadFields;
+        this.ignoredComplianceUsersForRead = ignoredComplianceUsersForRead;
+        this.watchedWriteIndicesPatterns = watchedWriteIndicesPatterns;
+        this.ignoredComplianceUsersForWrite = ignoredComplianceUsersForWrite;
+
+        this.readEnabledFields = watchedReadFields.entrySet().stream()
```

Comment:
what is the difference between this and https://github.com/opendistro-for-elasticsearch/security/pull/542/files#diff-0e04f575b10578faf6c21e321ea023b3R255-R262? Why are we handling them differently?

Reply:
The other place converts a comma separated list provided from elasticsearch settings yml to Map.
Eg: 
`twitter,id,message` is converted to 
```
{ 
   "twitter*": ["id", "message"]
}
```

And this converts map to WildcardMatcher patterns.
```
{
   <twitter wilcard pattern>: ["id", "message"]
}
```